packages/calcite-components/src/components/alert/alert.tsx

```diff
@@ -257,41 +229,107 @@ export class Alert implements OpenCloseComponent, LoadableComponent, T9nComponen
+                <slot name={SLOTS.title} />
+                <slot name={SLOTS.message} />
+                <slot name={SLOTS.link} />
+              </div>
             </div>
-          ) : null}
-          <div class="alert-content">
-            <slot name={SLOTS.title} />
-            <slot name={SLOTS.message} />
-            <slot name={SLOTS.link} />
           </div>
-          <div class={CSS.actionsEnd} hidden={!hasEndActions}>
-            {slotNode}
-          </div>
-          {this.queueLength > 1 ? queueCount : null}
-          {closeButton}
-          {open && !queued && autoClose ? <div class="alert-dismiss-progress" /> : null}
+          {hasEndActions && responsiveContainerWidth >= widthBreakpoints.small
+            ? this.renderActionsEnd()
+            : null}
+          {hasQueuedAlerts && responsiveContainerWidth >= widthBreakpoints.small
+            ? this.renderQueueCount()
+            : null}
+          {this.renderCloseButton()}
+          {(hasEndActions || hasQueuedAlerts) &&
+          responsiveContainerWidth < widthBreakpoints.small ? (
+            <div class={CSS.footer}>
+              {this.renderActionsEnd()}
+              {hasQueuedAlerts && responsiveContainerWidth < widthBreakpoints.small
+                ? this.renderQueueCount()
+                : null}
+            </div>
+          ) : null}
+          {open && !queued && autoClose ? <div class={CSS.dismissProgress} /> : null}
         </div>
       </Host>
     );
   }
 
+  private renderCloseButton(): VNode {
+    return (
+      <button
+        aria-label={this.messages.close}
+        class={CSS.close}
+        onClick={this.closeAlert}
+        type="button"
+        // eslint-disable-next-line react/jsx-sort-props -- ref should be last so node attrs/props are in sync (see https://github.com/Esri/calcite-design-system/pull/6530)
+        ref={(el) => (this.closeButton = el)}
+      >
+        <calcite-icon icon="x" scale={this.scale === "l" ? "m" : "s"} />
+      </button>
+    );
+  }
+
+  private renderQueueCount(): VNode {
+    const queueNumber = this.queueLength > 2 ? this.queueLength - 1 : 1;
+    const queueText = numberStringFormatter.numberFormatter.format(queueNumber);
+
+    return (
+      <div
+        class={{
+          [CSS.queueCount]: true,
+          [CSS.queueCountActive]: this.queueLength > 1,
+        }}
+        key="queue-count"
+      >
+        <calcite-chip scale={this.scale} value={queueText}>
+          {queueText}
+        </calcite-chip>
+      </div>
+    );
+  }
+
+  private renderActionsEnd(): VNode {
+    return (
+      <div class={CSS.actionsEnd}>
+        <slot name={SLOTS.actionsEnd} onSlotchange={this.actionsEndSlotChangeHandler} />
+      </div>
+    );
+  }
+
+  private renderIcon(icon: string): VNode {
+    return (
+      <div class={CSS.icon}>
+        <calcite-icon
+          flipRtl={this.iconFlipRtl}
+          icon={icon}
+          scale={this.scale === "l" ? "m" : "s"}
```

Comment:
Maybe a function for setting this scale value? This logic appears twice.

Reply:
I like the idea of having a util for internal icon scale adjustment. Especially, since it's sprinkled across many components. These seem to be the only exceptions I found:

* https://github.com/Esri/calcite-design-system/blob/8a575e451acb6b33d65124988b2e2f7ea8ec8b81/packages/calcite-components/src/components/popover/popover.tsx#L545
* https://github.com/Esri/calcite-design-system/blob/796bf90cacd68595a108f39e2a66bcc5d71de0e9/packages/calcite-components/src/components/modal/modal.tsx#L300

@ashetland @SkyeSeitz All internal icons scales should be following this (`m` icon for `scale="l"` and `s` icon for `scale="s"` + `scale="m"`) , right?

Reply:
Yep, that's correct.